activiti-steps/activiti-steps-assertions-api/src/main/java/org/activiti/steps/matchers/ProcessTaskMatchers.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.steps.matchers;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.activiti.api.process.model.events.BPMNActivityEvent;
+import org.activiti.api.process.model.events.BPMNActivityStartedEvent;
+import org.activiti.api.task.model.events.TaskRuntimeEvent;
+import org.activiti.api.task.runtime.events.TaskAssignedEvent;
+import org.activiti.api.task.runtime.events.TaskCreatedEvent;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class ProcessTaskMatchers {
+
+    private String taskName;
+
+    private ProcessTaskMatchers(String taskName) {
+        this.taskName = taskName;
+    }
+
+    public static ProcessTaskMatchers task(String taskName) {
+        return new ProcessTaskMatchers(taskName);
+    }
+
+    public OperationScopeMatcher hasBeenCreated() {
+
+        return (operationScope, events) -> {
+            hasBeenStarted().match(operationScope, events);
+            List<TaskCreatedEvent> taskCreatedEvents = events
+                    .stream()
+                    .filter(event -> TaskRuntimeEvent.TaskEvents.TASK_CREATED.equals(event.getEventType()))
+                    .map(TaskCreatedEvent.class::cast)
+                    .collect(Collectors.toList());
+            assertThat(taskCreatedEvents)
+                    .filteredOn(event -> event.getEntity().getProcessInstanceId().equals(operationScope.getProcessInstanceId()))
+                    .extracting(event -> event.getEntity().getName())
+                    .contains(taskName);
+
+        };
+    }
+
+    public OperationScopeMatcher hasBeenAssigned() {
+
+        return (operationScope, events) -> {
+            List<TaskAssignedEvent> taskAssignedEvents = events
+                    .stream()
+                    .filter(event -> TaskRuntimeEvent.TaskEvents.TASK_ASSIGNED.equals(event.getEventType()))
+                    .map(TaskAssignedEvent.class::cast)
+                    .collect(Collectors.toList());
+            assertThat(taskAssignedEvents)
+                    .filteredOn(event -> event.getEntity().getProcessInstanceId().equals(operationScope.getProcessInstanceId()))
+                    .extracting(event -> event.getEntity().getName())
+                    .contains(taskName);
+
+        };
+    }
+
+    private OperationScopeMatcher hasBeenStarted() {
```

Comment:
@erdemedeiros so suspended is missing here as well.. and cancelled??

Reply:
yes, they are missing in the current state.